app/poros/cafe_categorizer.rb

```diff
@@ -0,0 +1,56 @@
+class CafeCategorizer
+  class << self
+    def categorize(cafe)
+      case cafe.post_code[0..3]
+      when "LS1 "
+        categorize_ls1(cafe)
+      when "LS2 "
+        categorize_ls2(cafe)
+      else
+        cafe.category = 'other'
+      end
+      cafe.save
+    end
+
+    private
+
+    def categorize_ls1(cafe)
+      if cafe.number_of_chairs < 10
+        cafe.category = 'ls1 small'
+      elsif cafe.number_of_chairs < 100
+        cafe.category = 'ls1 medium'
+      elsif cafe.number_of_chairs >= 100
+        cafe.category = 'ls1 large'
+      end
+    end
+
+    def categorize_ls2(cafe)
+      if over_50_percentile?(cafe)
+        cafe.category = 'ls2 large'
+      else
+        cafe.category = 'ls2 small'
+      end
+    end
+
+    def percentile_50
+      @@percentile_50 ||= set_50_percentile
```

Comment:
Using memoization here allowed me to not calculate the 50th percentile for each cafe in the task